HunterJuneau/dine-and-dash
# Product Order API Calls and Subtraction Logic

## Description
Added the ability to Create(Add) a Product Order.
When Creating(adding) a Product Order, this logic will automatically subtract the Product Order Product Quantity from the Product's own Quantity. Thus, always keep the database updated with the most recent Product Quantity even when a user adds a Product to their cart.
If the Product is out of stock, a popup will appear when trying to create a Product Order telling the user the Product is out of stock. 
If the User tries to add a larger quantity of a Product than what is available, a popup will appear notifying the user of the current stock of the Product and how much of that Product they tried to add. 

## Related Issue

Close #51 

## Motivation and Context

- To construct an order, the user must add items to this order. That is what the Product Order is handling
- Needed to make sure that the Product Quantity can never be negative
- Needed to make sure that the User cannot order a Product if it is out of stock
- Need to make sure that the User cannot order a larger quantity of a Product that is available in stock
- Needed a way to always keep the database up to date with the current Product Quantity even when a User adds that item to their Order

## How Can This Be Tested?

To test this fully, first Get All Orders in Swagger. Then, Get All Products in Swagger. After that, begin to Add a Product Order. 
To test it as successful - 
Take one of the Order Ids and replace it with the existing Order Id in the Add Product Call.
Do the same with the Product Id. Make sure the Product you choose is not out of stock.
Add a Quantity lower or the same as the current Product Quantity. 
Then execute the Add. When done, you will see a Product Order in the response with all the details you have given the Add call and an Id created for that Product Order. 

To test out of stock Pop Up - 
Repeat steps, but choose a Product that has a Quantity of 0
You should see the pop up displaying that this Product is out of stock. 

To test User over-order Pop Up-
Repeat steps, but add a higher amount of that Product than what is the available Quantity in stock during the Add Call. You should see the Pop Up tell you the Quantity of the Product and the Quantity that you tried to add.  

## Types of changes
